dot/network/notifications.go

```diff
@@ -293,6 +295,12 @@ func (s *Service) sendHandshake(peer peer.ID, hs Handshake, info *notificationsP
 	// so we cannot have a method on peersData to lock and unlock the mutex
 	// from the map
 	peerMutex := info.peersData.getMutex(peer)
+	if peerMutex == nil {
+		// Note: the only place the mutex is deleted is when the peer disconnects.
+		// If it doesn't exist, the peer never connected either.
```

Comment:
If it doesn't exist, the peer either never connected or got disconnected, right?